node/node.go

```diff
@@ -950,23 +930,12 @@ func (n *Node) OnStart() error {
 			return fmt.Errorf("failed to start state sync: %w", err)
 		}
 	}
-	if n.embedIpfsNode {
-		// It is essential that we create a fresh instance of ipfs node on
-		// each start as internally the node gets only stopped once per instance.
-		// At least in ipfs 0.7.0; see:
-		// https://github.com/lazyledger/go-ipfs/blob/dd295e45608560d2ada7d7c8a30f1eef3f4019bb/core/builder.go#L48-L57
-		n.ipfsNode, err = createIpfsNode(n.config, n.areIpfsPluginsAlreadyLoaded, n.Logger)
-		if err != nil {
-			return fmt.Errorf("failed to create IPFS node: %w", err)
-		}
 
-		ipfsAPI, err := coreapi.NewCoreAPI(n.ipfsNode)
-		if err != nil {
-			return fmt.Errorf("failed to create an instance of the IPFS core API: %w", err)
-		}
-
-		n.consensusState.IpfsAPI = ipfsAPI
+	n.ipfsAPI, n.ipfsClose, err = n.ipfsAPIProvider(n.config, n.Logger)
```

Comment:
Are you planning to move this out of the OnStart method too?

Reply:
Not sure, I think running embedded IPFS in the last place makes sense